crates/sdk/src/masp.rs

```diff
@@ -2353,20 +2515,32 @@ impl<U: ShieldedUtils + MaybeSend + MaybeSync> ShieldedContext<U> {
             },
         );
         self.sync_status = ContextSyncStatus::Speculative;
+        for vk in &vks {
+            self.vk_heights.entry(*vk).or_default();
+        }
+        let ctx = ThreadHandle::new(self);
```

Comment:
instead of taking `self` directly, which is a `&mut` pointer, maybe steal the content of the pointer for the duration of these calls on `ctx`. then you don't have to wrap a `&mut` pointer with `Arc<Mutex<...>>` which is kind of hacky, and the `unsafe Send/Sync` impls will be safer.

Reply:
I mean, I can easily make this change, but it won't be any safer. A type `T` is `send` if and only if `&mut T` is `send`.  I guess the real benefit is that in this particular case `T` is not `clone` but `&mut T` is,

Reply:
Actually, this is basically impossible to change since I can't always get ownership of a shielded context from the NamadaImpl due to the RwLocks.